command/agent/config.go

```diff
@@ -467,9 +467,14 @@ type ServerConfig struct {
 
 	// EvalGCThreshold controls how "old" an eval must be to be collected by GC.
 	// Age is not the only requirement for a eval to be GCed but the threshold
-	// can be used to filter by age.
+	// can be used to filter by age. Please note that batch job evaluations are
+	// controlled by 'BatchEvalGCThreshold' instead.
 	EvalGCThreshold string `hcl:"eval_gc_threshold"`
 
+	// BatchEvalGCThreshold controls how "old" an evaluation must be to be eligible
+	// for GC if the eval belongs to a batch job.
+	BatchEvalGCThreshold string `hcl:"batch_eval_gc_threshold"`
```

Comment:
There's a `ParseDuration` call we'll need to make in `command/agent.go` as well. See the equivalent for `EvalGCThreshold` at [agent.go#L359-L365](https://github.com/hashicorp/nomad/blob/v1.4.3/command/agent/agent.go#L359-L365) for an example